prog/dftb+/lib_dftbplus/mainapi.F90

```diff
@@ -188,10 +191,433 @@ function nrOfAtoms()
   end function nrOfAtoms
 
 
+  !> Check that the order of speciesName remains constant
+  !> Keeping speciesNames constant avoids the need to reset:
+  !> atomEigVal, referenceN0, speciesMass and SK parameters 
+  function checkSpeciesNames(inputSpeciesName) result(tSpeciesNameChanged)
+    use dftbp_initprogram, only: speciesName
+    use dftbp_accuracy, only: mc
+
+    !> Labels of atomic species from external program
+    character(mc),  allocatable, intent(in) :: inputSpeciesName(:)
+    !> Has speciesName changed?
+    logical                                 :: tSpeciesNameChanged
+    
+    tSpeciesNameChanged = any(speciesName /= inputSpeciesName)
+ 
+  end function checkSpeciesNames
+
+  
+  !> When order of atoms changes, update arrays containing atom type indices,
+  !> and all subsequent dependencies. Updated data returned via use statements
+  subroutine updateDataDependentOnSpeciesOrdering(env, inputSpecies)
+    !TODO(Alex) These will go to the top of the module
```

Comment:
Move "use module, only:" to top of current module? I like declaring per subroutine to limit scope but still appears to be declare once per .f90 file

Reply:
We generally place all module imports to the top level, so please do so.